packages/agreement-process/test/activateAgreement.test.ts

```diff
@@ -0,0 +1,2171 @@
+      const consumerId = generateId<TenantId>();
+      const authData = getRandomAuthData(consumerId);
+
+      const agreement: Agreement = {
+        ...getMockAgreement(),
+        state: randomArrayItem(
+          Object.values(agreementState).filter(
+            (state) => !agreementActivableStates.includes(state)
+          )
+        ),
+        consumerId,
+      };
+      await addOneAgreement(agreement);
+      await expect(
+        agreementService.activateAgreement(agreement.id, {
+          authData,
+          serviceName: "",
+          correlationId: "",
+          logger: genericLogger,
+        })
+      ).rejects.toThrowError(
+        agreementNotInExpectedState(agreement.id, agreement.state)
+      );
+    });
+
+    it("should throw an eServiceNotFound error when the EService does not exist", async () => {
+      const consumerId = generateId<TenantId>();
+      const authData = getRandomAuthData(consumerId);
+
+      const agreement: Agreement = {
+        ...getMockAgreement(),
+        state: randomArrayItem(
+          agreementActivableStates.filter((s) => s !== agreementState.pending)
+        ),
+        consumerId,
+      };
+      await addOneAgreement(agreement);
+      await expect(
+        agreementService.activateAgreement(agreement.id, {
+          authData,
+          serviceName: "",
+          correlationId: "",
+          logger: genericLogger,
+        })
+      ).rejects.toThrowError(eServiceNotFound(agreement.eserviceId));
+    });
+
+    it("should throw a descriptorNotFound error when the Descriptor does not exist", async () => {
+      const consumerId = generateId<TenantId>();
+      const producerId = generateId<TenantId>();
+      const authData = getRandomAuthData(producerId);
+
+      const eservice: EService = {
+        ...getMockEService(),
+        producerId,
+      };
+      const agreement: Agreement = {
+        ...getMockAgreement(),
+        eserviceId: eservice.id,
+        consumerId,
+        state: randomArrayItem(agreementActivableStates),
+        producerId,
+      };
+
+      await addOneEService(eservice);
+      await addOneAgreement(agreement);
+
+      await expect(
+        agreementService.activateAgreement(agreement.id, {
+          authData,
+          serviceName: "",
+          correlationId: "",
+          logger: genericLogger,
+        })
+      ).rejects.toThrowError(
+        descriptorNotFound(agreement.eserviceId, agreement.descriptorId)
+      );
+    });
+
+    it("should throw a descriptorNotInExpectedState error when the Descriptor is not in an expected state", async () => {
+      const consumerId = generateId<TenantId>();
+      const producerId = generateId<TenantId>();
+      const authData = getRandomAuthData(producerId);
+
+      const descriptor: Descriptor = {
+        ...getMockDescriptorPublished(),
+        state: randomArrayItem(
```

Comment:
since we have a finite number of values, instead of random item we should test all cases